tests/pytests/unit/modules/test_pcs.py

```diff
@@ -0,0 +1,314 @@
+from salt.utils import extend
+import pytest
+import salt.modules.pcs as pcs
+from tests.support.mock import MagicMock, patch
+
+
+
+class TstData:
+    def __init__(self):
+        self.nodea = "nodea"
+        self.nodeb = "nodeb"
+        self.nodes = [self.nodea, self.nodeb]
+        self.extra_args = ["test", "extra", "args"]
+        self.cluster_name = "testcluster"
+        self.username = "hacluster"
+        self.password = "hacluster"
+        self.cib_filename = "/tmp/cib.xml"
+
+
+@pytest.fixture
+def test_data():
+    return TstData()
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {
+        pcs: {"__salt__": {"pkg.version": MagicMock()}},
+    }
+
+@pytest.mark.parametrize("ver_cmp_ret,old_ver", [(1, False), (0, True)])
+def test_auth(ver_cmp_ret, old_ver, test_data):
+    """
+    Test for authorising hosts on cluster
+    """
+    exp_cmd = ["pcs"]
+    if old_ver:
+        exp_cmd.extend(["cluster", "auth"])
+    else:
+        exp_cmd.extend(["host", "auth"])
+
+    exp_cmd.extend(["-u", test_data.username, "-p" , test_data.password])
+    exp_cmd.extend(test_data.extra_args)
+    exp_cmd.extend([test_data.nodea, test_data.nodeb]) 
+    
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(
+        pcs.__salt__,
+        {
+            "cmd.run_all": mock_cmd,
+            "pkg.version_cmp": MagicMock(return_value=ver_cmp_ret),
+        },
+    )
+
+    with patch_salt:
+        pcs.auth(
+            test_data.nodes,pcsuser = test_data.username, pcspasswd = test_data.password, extra_args=test_data.extra_args
+        )
+    assert mock_cmd.call_args_list[0][0][0] == exp_cmd
+
+def test_is_auth_old(test_data):
+    """
+    Test for checking it nodes are authorised. 
+    """
+    exp_cmd = ["pcs", "cluster", "auth"]
+    exp_cmd.extend(test_data.nodes)
+    mock_cmd = MagicMock()
+    patch_salt = patch.dict(
+        pcs.__salt__,
+        {
+            "cmd.run_all": mock_cmd,
+            "pkg.version_cmp": MagicMock(return_value=0),
+        },
+    )
+
+    with patch_salt:
+        pcs.is_auth(test_data.nodes, pcsuser=test_data.username, pcspasswd=test_data.password)
+    
+    print("Exp command: ", exp_cmd)
```

Comment:
Can you remove this print and the one below.

Reply:
Done